Deleted functionality of the cache API config.

RamosFe opened the pull request:
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/22372|



## Description

Deleted all functionality and tests related to the `cache` API configuration.

fdalmaup reviewed (changes requested):
- Given the changes this PR introduces, at least the AITs `test_cluster_endpoints.tavern.yaml` and `test_manager_endpoints.tavern.yaml` should be locally run (**also check the failed tests in the Checks section**). 
- Also, a manual test to show the displayed warning should be added as proof of the behavior when a user still adds the deprecated configuration.

RamosFe commented:
## Update
Added IT results in https://github.com/wazuh/wazuh/issues/22372#issuecomment-2048328723.

RamosFe commented on api/scripts/wazuh-apid.py:
```diff
@@ -11,6 +11,8 @@
 import warnings
 
 SSL_DEPRECATED_MESSAGE = 'The `{ssl_protocol}` SSL protocol is deprecated.'
+CACHE_DELETED_MESSAGE = 'The `cache` API configuration option no longer take effect since {release} and will ' \
```
Fixed in e3d002f23db667b51943762712f1073e3b7bb82c.

fdalmaup reviewed (approved):
LGTM!

Selutario reviewed (changes requested):
The changes look good, but there are some more things to remove:

1. **[API config IT (cache)](https://github.com/wazuh/wazuh/tree/remove/22372-delete-api-cache-config/tests/integration/test_api/test_config/test_cache)**. The test should also be removed.
2. [**Configuration model**](https://github.com/wazuh/wazuh/blob/remove/22372-delete-api-cache-config/api/api/models/configuration_model.py). It was deprecated in 4.0.4 and has been maintained ever since for no apparent reason. It is not being used anywhere and contains references to `cache` configuration, which will no longer exist. We can delete the entire file.
3. [**Documentation**](https://documentation.wazuh.com/current/user-manual/api/configuration.html#cache). The cache configuration doc should also be removed in a new PR targeting #22372.

RamosFe commented:
## Update
- Deleted IT in 8da4c04b1fc482ae7bee197775fc684073780ad5.
- Deleted Model in be5ccf2a4111dbc47a63edb37a39a659301daaee.
- The deprecated tag was added in https://github.com/wazuh/wazuh-documentation/blob/e0847c666f685c63d77e07261aedebe725b0b7a9/source/user-manual/api/configuration.rst?plain=1#L280-L292 (Version `4.8.0`)

Selutario reviewed (approved):
LGTM
